docs/src/main/asciidoc/spring-boot-properties.adoc

```diff
@@ -0,0 +1,373 @@
+
+[source,java]
+----
+package org.acme.spring.boot.properties;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/greeting")
+public class GreetingResource {
+
+    @Inject
+    GreetingProperties properties;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return properties.text;
+    }
+}
+----
+
+Run the tests to verify that application still functions correctly.
+
+== Package and run the application
+
+Run the application with: `./mvnw compile quarkus:dev`.
+Open your browser to http://localhost:8080/greeting.
+
+Changing the configuration file is immediately reflected.
+
+As usual, the application can be packaged using `./mvnw clean package` and executed using the `-runner.jar` file.
+You can also generate the native executable with `./mvnw clean package -Pnative`.
+
+== Default values
+
+Now let's add a suffix for a greeting which we'll set to have a default value.
+
+
+Properties with default values can be configured in a configuration file just like any other property.
+However, the default value will be used if the property was not defined in a configuration file.
+
+Go ahead and add the new field to the `GreetingProperties` class:
+
+[source,java]
+----
+package org.acme.spring.boot.properties;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties("greeting")
+public class GreetingProperties {
+
+    public String text;
+
+    public String suffix = "!";
+}
+----
+
+And update the `GreetingResource` and its test `GreetingResourceTest`:
+
+[source,java]
+----
+package org.acme.spring.boot.properties;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/greeting")
+public class GreetingResource {
+
+    @Inject
+    GreetingProperties properties;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return properties.text + properties.suffix;
+    }
+}
+----
+
+[source,java]
+----
+package org.acme.spring.boot.properties;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+public class GreetingResourceTest {
+
+    @Test
+    public void testHelloEndpoint() {
+        given()
+          .when().get("/greeting")
+          .then()
+             .statusCode(200)
+             .body(is("hello!"));
+    }
+}
+----
+
+Run the tests to verify the change.
+
+== Optional values
+
+Properties with optional values are the middle-ground between standard and properties with default values.
+While a missing property in a configuration file will not cause your application to fail it will neither have a value set to it.
```

Comment:
```suggestion
While a missing property in a configuration file will not cause your application to fail, it will nevertheless not have a value set.
```